Zend/zend.h

```diff
@@ -69,7 +69,7 @@
 #define ZEND_TSRMLS_CACHE
 #endif
 
-ZEND_TSRMLS_CACHE_EXTERN()
+TSRMLS_MAIN_CACHE_EXTERN()
```

Comment:
I don't really get how this is supposed to work. zend.h will be included in shared extension files as well, so it will have both an extern declaration with `tls_model` and another one without (the ZEND_TSRMLS_CACHE_EXTERN in the extension header). Wouldn't that cause issues?

Reply:
On Linux this works, but yeah. It would be better to wrap this declaration with some ``#ifdef``.

Reply:
@nikic the last patch should fix the ambiguity